src/webgl/p5.RendererGL.Immediate.js

```diff
@@ -113,6 +121,19 @@ p5.RendererGL.prototype.vertex = function(x, y) {
   const vert = new p5.Vector(x, y, z);
   this.immediateMode.geometry.vertices.push(vert);
   this.immediateMode.geometry.vertexNormals.push(this._currentNormal);
+  if (this._useUserAttributes){ 
+    const geom = this.immediateMode.geometry;
+    const verts = geom.vertices;
+    Object.entries(this.userAttributes).forEach(([name, data]) => {
+      const size = data.length ? data.length : 1;
+      if (verts.length > 0 && !geom.hasOwnProperty(name)) {
+        for (let i = 0; i < verts.length - 1; i++) {
```

Comment:
Not urgent because this definitely works, but similar to what you did above in the geometry builder, you could maybe replace this for loop with something like `Array(numMissingValues).fill(0)`?

Reply:
I've just added a [size parameter](https://github.com/processing/p5.js/pull/7276/commits/6ec85c118074d4b03f56504ab2535b8c2f7d8c56 to `Geometry.setAttribute()` to make this easier.

)```js
setAttribute(attributeName, data, size = data.length ? data.length : 1){}
```